source/rules/literals.rs

```diff
@@ -42,25 +42,26 @@ use nom::{
     hex_digit,
     oct_digit
 };
-use std::str;
 use std::str::FromStr;
+use std::str;
+use super::super::ast::Literal;
 
 named!(
-    pub null< Option<()> >,
+    pub null<Literal>,
     map_res!(
         tag!("null"),
```

Comment:
I'm afraid this will be case sensitive. Should we use `alt!(tag!("null") | tag!("NULL"))`